Added basic project structure:

j-fryc opened the pull request:
- implemented repository pattern to handle data access
- added project configuration
- added example endpoint

j-fryc committed:
Added basic project structure:
- implemented repository pattern to handle data access
- added project configuration
- added example endpoint

kgarbacinski commented on Dockerfile:
```diff
@@ -0,0 +1,7 @@
+COPY requirements.txt entry_point_fastapi.sh .env ./
+RUN pip install --no-cache-dir -r requirements.txt
+COPY ./app /app
+EXPOSE 8002
```
Use 8100, 8200 instead of 8001, 8002 etc.

kgarbacinski commented on app/config.py:
```diff
@@ -0,0 +1,14 @@
+class Settings(BaseSettings):
+    secret_key: str
+
+    model_config = SettingsConfigDict(env_file="../../.env")
```
you don't need .env when running the app in a container; set ENVs for the app in docker-compose.yml

kgarbacinski commented on app/db_handler/db_handler.py:
```diff
@@ -0,0 +1,28 @@
+
+class Database:
+    def __init__(self, connection_string: str, pool_size: int = 20, max_overflow: int = 0):
+        self.sql_url = connection_string
+        self.connect_args = {"check_same_thread": False}
+        self._engine = create_engine(
+            self.sql_url,
+            connect_args=self.connect_args,
+            pool_size=pool_size,
+            max_overflow=max_overflow
+        )
```
use async session for the connection with database

kgarbacinski commented on app/main.py:
```diff
@@ -0,0 +1,28 @@
+
+@app.on_event("startup")
+async def startup():
+    db = Database(connection_string='sqlite:///app/data/foo.db')
+    db.create_db_and_tables()
+    app.state.db = db
```
use postgresql

j-fryc committed:
Changes after CR

j-fryc committed:
Implemented CUD logic with 3 layers logic:
- Added repository layer to handle db operations
- Added service layer to handler bussines logic
- Adjusted presentation layer by removing bussines logic and moving it to service layer
